locust/argument_parser.py

```diff
@@ -369,6 +371,66 @@ def parse_locustfile_option(args=None) -> list[str]:
     return parsed_paths
 
 
+# A hack for setting up an action that raises ArgumentError with configurable error messages.
+# This is meant to be used to immediately block use of deprecated arguments with some helpful messaging.
+
+
+def raise_argument_type_error(err_msg):
+    class ErrorRaisingAction(configargparse.Action):
+        def __call__(self, parser, namespace, values, option_string=None):
+            raise configargparse.ArgumentError(self, err_msg)
+
+    return ErrorRaisingAction
+
+
+# Definitions for some "types" to use with the arguments
+
+
+def timespan(time_str):
```

Comment:
Cool stuff! Probably wanna add a type hint for the return type at least?

If you do, does the hint carry over to when people do environment.parsed_options.stop_timeout? That'd be cool...